pkg/security/secl/model/unmarshallers_linux.go

```diff
@@ -466,53 +466,33 @@ func (m *Mount) UnmarshalBinary(data []byte) (int, error) {
 	}
 
 	m.MountID = m.RootPathKey.MountID
-	m.Origin = MountOriginEvent
-
-	return 56, nil
-}
 
-// UnmarshalBinary unmarshalls a binary representation of itself
-func (e *MountEvent) UnmarshalBinary(data []byte) (int, error) {
-	return UnmarshalBinary(data, &e.SyscallEvent, &e.SyscallContext, &e.Mount)
-}
+	bitfield := binary.NativeEndian.Uint32(data[24:28])
+	m.Visible = bitfield&0b01 > 0
+	m.Detached = bitfield&0b10 > 0
 
-// ToMount transforms an fsmount event to be used with the mount resolver
-func (e *FsmountEvent) ToMount() *Mount {
-	return &Mount{
-		Origin:      MountOriginFsmount,
-		MountID:     e.MountID,
-		Device:      e.Device,
-		RootPathKey: e.RootPathKey,
-		RootStr:     "/",
-	}
+	return 64, nil
 }
 
 // UnmarshalBinary unmarshalls a binary representation of itself
-func (e *FsmountEvent) UnmarshalBinary(data []byte) (int, error) {
-	read, err := e.SyscallEvent.UnmarshalBinary(data)
-	if err != nil {
-		return 0, err
-	}
-	data = data[read:]
-	n, err := e.RootPathKey.UnmarshalBinary(data)
+func (e *MountEvent) UnmarshalBinary(data []byte) (int, error) {
+	n, err := UnmarshalBinary(data, &e.SyscallEvent, &e.SyscallContext, &e.Mount)
 	if err != nil {
-		return read, err
+		return n, err
 	}
-
-	read += n
 	data = data[n:]
+	origin := binary.NativeEndian.Uint32(data[0:4])
```

Comment:
Is it possible to check the bounds of data before reading `origin` to avoid any potential crashes?